client/subscription-edit-page/index.tsx

```diff
@@ -0,0 +1,280 @@
+		);
+	}
+
+	const result = await response.json();
+	return result.data as FetchUserTokensResponse;
+};
+
+/**
+ * Renders a payment method select or loading indicator.
+ *
+ * @param {PaymentMethodSelectProps} props The props for the payment method select.
+ * @param {string} props.inputName The name attribute for the select element.
+ * @param {number} props.value The currently selected payment method token ID.
+ * @param {(value: number) => void} props.onChange Callback when the selected value changes.
+ * @param {number} props.userId The ID of the customer whose payment methods to display.
+ * @param {UserTokenCache} props.cache The cache containing user payment token data.
+ * @return {JSX.Element} The payment method select or loading indicator.
+ */
+export const PaymentMethodSelect = ( {
+	inputName,
+	value,
+	onChange,
+	userId,
+	cache,
+}: PaymentMethodSelectProps ) => {
+	/**
+	 * Generate options for the select.
+	 */
+	const options: JSX.Element[] = [];
+	if ( userId > 0 ) {
+		const entry = cache.getUserEntry( userId );
+		if ( undefined === entry || entry.loading ) {
+			return <>{ __( 'Loading…', 'woocommerce-payments' ) }</>;
+		} else if ( entry.loadingError ) {
+			return <strong>{ entry.loadingError }</strong>;
+		}
+
+		if ( ! cache.userHasToken( userId, value ) ) {
+			options.push(
+				<option value={ 0 } key={ 'select' } disabled>
+					{ __(
+						'Please select a payment method',
+						'woocommerce-payments'
+					) }
+				</option>
+			);
+		}
+
+		entry.tokens.forEach( ( token ) => {
+			options.push(
+				<option value={ token.tokenId } key={ token.tokenId }>
+					{ token.displayName }
+				</option>
+			);
+		} );
+	} else {
+		options.push(
+			<option value={ 0 } key={ 'no-customer' } disabled>
+				{ __(
+					'Please select a customer first',
+					'woocommerce-payments'
+				) }
+			</option>
+		);
+	}
+
+	return (
+		// eslint-disable-next-line
+		<select
+			name={ inputName }
+			value={ value }
+			onChange={ ( event ) =>
+				onChange( parseInt( event.target.value, 10 ) )
+			}
+		>
+			{ options }
+		</select>
+	);
+};
+
+/**
+ * Setup the payment method select for a given element.
+ *
+ * @param {HTMLSpanElement} element The <span> where the payment method select should be rendered.
+ * @param {UserTokenCache} cache The cache of user tokens.
+ * @return {void}
+ */
+const setupPaymentSelector = (
+	element: HTMLSpanElement,
+	cache: UserTokenCache
+): void => {
+	const data = JSON.parse(
+		element.getAttribute( 'data-wcpay-pm-selector' ) || '{}'
+	) as WCPayPMSelectorData;
+
+	// Use the values from the data instead of input to ensure correct types.
+	let userId = data.userId ?? 0;
+	let value = data.value ?? 0;
+
+	// Initial population.
+	if ( userId ) {
+		cache.add( userId, data.tokens ?? [] );
+	}
+
+	// In older Subscriptions versions, there was just a simple input.
+	const input = element.querySelector( 'select,input' ) as
+		| HTMLSelectElement
+		| HTMLInputElement
+		| null;
```

Comment:
(minor) similar to above - we can check the returned type, instead of forcing it.

```suggestion
	const input = element.querySelector( 'select,input' );
	if ( ! input ) {
		return;
	}
	if (
		! (
			input instanceof HTMLSelectElement ||
			input instanceof HTMLInputElement
		)
	) {
		return;
	}
```

Reply:
Addressed in https://github.com/Automattic/woocommerce-payments/pull/11143/commits/4043b4914c6f4e20531c154d2396f124212d7fe1